test/dispute.js

```diff
@@ -0,0 +1,315 @@
+
+import Merkelizer from '../utils/Merkelizer';
+import DisputeMock from '../utils/DisputeMock';
+import OffchainStepper from '../utils/OffchainStepper';
+
+// for additional logging
+const DEBUG = false;
+const OP = require('./helpers/constants');
+
+function debugLog (...args) {
+  if (DEBUG) {
+    console.log(...args);
+  }
+}
+
+function submitProofHelper (dispute, code, computationPath) {
+  const prevOutput = computationPath.left.executionState.output;
+  const execState = computationPath.right.executionState;
+  const input = execState.input;
+
+  const proofs = {
+    stackHash: Merkelizer.stackHash(
+      prevOutput.stack.slice(0, prevOutput.stack.length - input.compactStack.length)
+    ),
+    memHash: execState.isMemoryRequired ? '' : Merkelizer.memHash(prevOutput.mem),
+    dataHash: execState.isCallDataRequired ? '' : Merkelizer.dataHash(input.data),
+  };
+
+  return dispute.submitProof(
+    proofs,
+    {
+      // TODO: compact {code,returnData}, support for accounts
+      // code: input.code,
+      code: code,
+      data: execState.isCallDataRequired ? input.data : '',
+      stack: input.compactStack,
+      mem: execState.isMemoryRequired ? input.mem : '',
+      returnData: input.returnData,
+      pc: input.pc,
+      logHash: input.logHash,
+      gasRemaining: input.gasRemaining,
+    }
+  );
+}
+
+async function disputeGame (code, solverSteps, challengerSteps, expectedWinner, expectedError) {
+  try {
+    const stepper = OffchainStepper;
+    const solverMerkle = new Merkelizer().run(solverSteps);
+    const challengerMerkle = new Merkelizer().run(challengerSteps);
+
+    if (DEBUG) {
+      debugLog('solver depth=' + solverMerkle.depth);
+      solverMerkle.printTree();
+      debugLog('challenger depth=' + challengerMerkle.depth);
+      challengerMerkle.printTree();
+    }
+
+    let solverComputationPath = solverMerkle.root;
+    let challengerComputationPath = challengerMerkle.root;
+
+    // TODO: handle the bigger case too
+    if (solverMerkle.depth < challengerMerkle.depth) {
+      challengerComputationPath = challengerMerkle.tree[solverMerkle.depth][0];
```

Comment:
this takes a subtree from the bigger tree? which is a subset of the execution path (not all leaves), right? wouldn't then the last step be different from REVERT/RETURN?